scapy/layers/inet6.py

```diff
@@ -1084,14 +1084,14 @@ def defragment6(pktlist):
     Crap is dropped. What lacks is completed by 'X' characters.
     """
 
-    l = filter(lambda x: IPv6ExtHdrFragment in x, pktlist) # remove non fragments
+    l = [x for x in pktlist if IPv6ExtHdrFragment in x] # remove non fragments
```

Comment:
Then that's not a `PacketList` object and the variable name is misleading... Maybe it should be renamed as `packets`?